RFC: stdenv: enable parallel builds by default in make-derivation.nix

trofi opened the pull request:
Vast majority of packages can be built in parallel without problems.
My initial goal was to speed up initial bootstrap slightly where
packages like `bison` or `coreutils` happened to build sequentially
kist because it's a default.

This change flips the default to parallel builds for make-based
derivations. We expect some build failure fallou which in worst case
should be easy to fix with:

    # <why it fails>
    enableParallelBuilding = false;

Issue: https://github.com/NixOS/nixpkgs/issues/142338

trofi committed:
RFC: stdenv: enable parallel builds by default in make-derivation.nix

Vast majority of packages can be built in parallel without problems.
My initial goal was to speed up initial bootstrap slightly where
packages like `bison` or `coreutils` happened to build sequentially
kist because it's a default.

This change flips the default to parallel builds for make-based
derivations. We expect some build failure fallou which in worst case
should be easy to fix with:

    # <why it fails>
    enableParallelBuilding = false;

Issue: https://github.com/NixOS/nixpkgs/issues/142338

dasj19 commented:
What does it take to get this merged? I have 32 cores and when compiling nix packages it just uses one of them at a time.
Anything left to do?

trofi commented:
https://github.com/NixOS/nixpkgs/issues/142338 has a list of packages to fix before we can merge it. I'll update it with local list of hacks as well.

trofi commented:
Rebasing against `master`'s hydra run: https://hydra.nixos.org/eval/1742076 (75b305d5b9fa462297e067d6a917f616445bb7fb). I'll ask a separate hydra job for this PR to get more failures before we merge it to staging.

trofi committed:
stdenv: DO NOT MERGE: require "big-parallel" machines to run most derivations

Avoid jobs with small parallelism on hydra (-j2) and try hard to
use only bug machines. It will make the run slower, but will hopefully
get more useful results.

trofi commented:
Defaults change was rejected in https://github.com/NixOS/nixpkgs/issues/142338
